datadotworld/client/api.py

```diff
@@ -441,6 +445,314 @@ def download_datapackage(self, dataset_key, dest_dir):
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def sql_post(self, dataset_key, query):
+        """Executes SQL queries against a dataset via POST
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        query : str
+            SQL query
+        include_table_schema : bool
+            Flags indicating to include table schema in the response
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.sql_post('username/test-dataset', query)
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._sql_api.sql_post(owner_id, dataset_id, query)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    # Sparql Operations
+
+    def sparql_get(self, dataset_key, query):
+        """Executes SPARQL queries against a dataset via GET
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        query : str
+            SPARQL query
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.sparql_get('username/test-dataset', query)
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._sparql_api.sparql_get(owner_id, dataset_id, query)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def sparql_post(self, dataset_key, query):
+        """Executes SPARQL queries against a dataset via POST
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        query : str
+            SPARQL query
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> api_client.sparql_post('username/test-dataset', query)
+        """
+        owner_id, dataset_id = parse_dataset_key(dataset_key)
+        try:
+            return self._sparql_api.sparql_post(owner_id, dataset_id, query)
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    # Download Operations
+
+    def download_dataset(self, dataset_key):
+        """Return a .zip containing all files within the dataset as uploaded.
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
```

Comment:
What does this function return?

Reply:
`None` here doesn't seem right either. A file-like object might be right here again, but alternatively, you may request that the user provides a path where the downloaded file is to be saved.